bindings/node/test/clientEncryption.test.js

```diff
@@ -815,4 +828,220 @@ describe('ClientEncryption', function () {
+      it('when options.encryptedFields.fields is not an array no key generation is performed', async () => {
+        const createCollectionSpy = sinon.spy(db, 'createCollection');
+        const createDataKeySpy = sinon.spy(clientEncryption, 'createDataKey');
+        await clientEncryption.createEncryptedCollection(db, collectionName, {
+          createCollectionOptions: { encryptedFields: { fields: 'not an array' } }
+        });
+
+        expect(createDataKeySpy.callCount).to.equal(0);
+        const options = createCollectionSpy.getCall(0).args[1];
+        expect(options).to.deep.equal({ encryptedFields: { fields: 'not an array' } });
+      });
+
+      it('when options.encryptedFields.fields elements are not objects they are passed along to createCollection', async () => {
+        const createCollectionSpy = sinon.spy(db, 'createCollection');
+        const keyId = new Binary(Buffer.alloc(16, 0));
+        const createDataKeyStub = sinon.stub(clientEncryption, 'createDataKey').resolves(keyId);
+        await clientEncryption.createEncryptedCollection(db, collectionName, {
+          createCollectionOptions: {
+            encryptedFields: { fields: ['not an array', { keyId: null }, { keyId: {} }] }
+          }
+        });
+
+        expect(createDataKeyStub.callCount).to.equal(1);
+        const options = createCollectionSpy.getCall(0).args[1];
+        expect(options).to.deep.equal({
+          encryptedFields: { fields: ['not an array', { keyId: keyId }, { keyId: {} }] }
+        });
+      });
+    });
+
+    context('when subroutines throw the error is wrapped', () => {
+      it('throws MongoCryptCreateDataKeyError if createDataKey rejects', async () => {
+        const customError = new Error('evil!');
+        const stub = sinon.stub(clientEncryption, 'createDataKey');
+        const keyId = new Binary(Buffer.alloc(16, 0), 4);
+        stub.onCall(0).resolves(keyId);
+        stub.onCall(1).rejects(customError);
+        stub.onCall(2).resolves(keyId);
+
+        const createCollectionOptions = {
+          encryptedFields: { fields: [{}, {}, { keyId: 'cool id!' }] }
+        };
+        const error = await clientEncryption
+          .createEncryptedCollection(db, collectionName, {
+            provider: 'local',
+            createCollectionOptions
+          })
+          .catch(error => error);
+
+        expect(error).to.be.instanceOf(MongoCryptCreateDataKeyError);
+        expect(error.cause).to.equal(customError);
+
+        expect(error.encryptedFields).property('fields').that.is.an('array');
+        expect(error.encryptedFields.fields).to.have.lengthOf(
+          createCollectionOptions.encryptedFields.fields.length
+        );
+        expect(error.encryptedFields.fields).to.have.nested.property('[0].keyId', keyId);
+        expect(error.encryptedFields.fields).to.not.have.nested.property('[1].keyId');
+        expect(error.encryptedFields.fields).to.have.nested.property('[2].keyId', 'cool id!');
```

Comment:
From the test output, it's not clear that we're actually testing that the `encryptedFields` is present on the error:

```
      when subroutines throw the error is wrapped
        ✔ throws MongoCryptCreateDataKeyError if createDataKey rejects
        ✔ throws MongoCryptCreateEncryptedCollectionError if createCollection rejects
      when there are nullish keyIds in the encryptedFields.fields array
        ✔ does not mutate the input fields array when generating data keys
        ✔ generates dataKeys for all null keyIds in the fields array
```

I suggest breaking this up into context + it blocks, so the output is clear.  As an example, the following output more closely documents what we expect to happen:

when createDataKeyRejects
- a MongoCryptCreateDataKeyError is thrown with error message < whatever >
- the `MongoCryptCreateDataKeyError` contains a partially filled encrypted fields map
- the `MongoCryptCreateDataKeyError` contains the first error in the array of rejected promises

Reply:
this test breakup would apply to all the tests that contain assertions on multiple behaviors

Reply:
tests reorganized